# CDRIVER-6367: CMake 4.4 treats CMAKE_FLAGS "-Werror -D..." as a warning
# category name ("error -D..."), which aborts configure.
# https://github.com/mongodb/mongo-c-driver/commit/ea1b54791ec2f1aafb665f98af6cc9dd21116782
diff --git a/src/libmongoc/CMakeLists.txt b/src/libmongoc/CMakeLists.txt
--- a/src/libmongoc/CMakeLists.txt
+++ b/src/libmongoc/CMakeLists.txt
@@ -450,9 +450,12 @@
    }
    ")
 
-   TRY_COMPILE (RES ${CMAKE_CURRENT_BINARY_DIR}
-   ${CMAKE_CURRENT_BINARY_DIR}/accept_test${VAR}.c CMAKE_FLAGS
-   "-Werror -DCMAKE_CXX_LINK_EXECUTABLE='echo not linking now...'" OUTPUT_VARIABLE LOG2)
+   TRY_COMPILE (RES
+      "${CMAKE_CURRENT_BINARY_DIR}"
+      SOURCES "${CMAKE_CURRENT_BINARY_DIR}/accept_test${VAR}.c"
+      CMAKE_FLAGS "-DCOMPILE_DEFINITIONS=-Werror" "-DCMAKE_CXX_LINK_EXECUTABLE='echo not linking now...'"
+      OUTPUT_VARIABLE LOG2
+   )
 
    if (RES)
       message (
